test/integration/conftest.py

```diff
@@ -38,8 +38,13 @@ def api_isolation(request) -> Iterator[ApiTestEnvironment]:
     utils.close_environments(environment)
 
 
+CliContextProvider = Callable[
```

Comment:
This doesn't work too well, you need to use NewType, because otherwise the type checker can't work with that